zstacklib/zstacklib/utils/shell.py

```
@@ -16,19 +19,54 @@ def _linux_close_fds(self, but):
+    """
+    try:
+        os.killpg(os.getpgid(process.pid), signal.SIGKILL)
+    except OSError:
+        try:
+            process.kill()
+        except OSError:
+            pass
+
+    try:
+        process.wait(timeout=5)
+    except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)